lib/android_apk/signature_verifier.rb

```diff
@@ -27,47 +25,38 @@ module SignatureVerifier
       ]
     ].freeze
 
+    # @param [String] filepath to an apk file
+    # @param [Integer, String] min_sdk_version of an apk
+    # @return [Array<Hash<String => Any>>] sdk-ranged certificate information array order by min_sdk_version asc
     module_function def verify(filepath:, min_sdk_version:)
       min_sdk_version = min_sdk_version.to_i
 
       SCHEME_CASE_INPUTS.each_with_object([]) do |versions, constraints|
         min_sdk, max_sdk, = versions
-        min_sdk_version_for_verification = [min_sdk, min_sdk_version].max
-
         next unless min_sdk_version <= max_sdk
 
-        signer_hunks = ::AndroidApk::ApkSigner.verify(
+        signer_hunks = ::AndroidApk::ApkSigner.print_certs(
           filepath: filepath,
-          min_sdk_version: min_sdk_version_for_verification,
+          min_sdk_version: [min_sdk, min_sdk_version].max,
           max_sdk_version: max_sdk
         )
 
-        if signer_hunks.empty?
-          constraints.push(
+        next if signer_hunks.empty?
+
+        signer_hunks.each do |sdk_versions, signer_lines|
+          # { "sha1" => ..., ... }
+          signature = signer_lines.each_with_object(
             {
-              "min_sdk_version" => min_sdk_version_for_verification,
-              "max_sdk_version" => max_sdk,
-              ::AndroidApk::SignatureDigest::MD5 => nil,
```

Comment:
Don't create a special range for *unsigned* because no active range has the same meaning.